linera-core/src/client/mod.rs

```diff
@@ -1314,14 +1349,14 @@ where
         &self,
         chain_id: ChainId,
         block_batch: &mut Vec<BlockHeight>,
-        tracker: &mut MultichainTracker,
+        tracker: &Arc<RwLock<MultichainTracker>>,
     ) -> Result<(), NodeError> {
```

Comment:
Would it make sense to make most of this a method of `MultichainTracker`, that instead of a `ChainClient` directly takes the local `next_block_height` as an argument?